Meter.h

```diff
@@ -96,6 +107,27 @@ typedef enum {
    LAST_METERMODE
 } MeterModeId;
 
+#define IS_POWER_OF_2(x) ((x) > 0 && !((x) & ((x) - 1)))
+
+#ifndef __has_builtin
+# define __has_builtin(x) 0
+#endif
+#if (__has_builtin(__builtin_clz) || \
+    ((__GNUC__ > 3) || (__GNUC__ == 3 && __GNUC_MINOR__ >= 4)))
+# define HAS_BUILTIN_CLZ 1
+# define HAS_ILOG2 1
+/*
+ * ilog2(x): base-2 logarithm of an unsigned integer x, rounded down, but
+ *           ilog2(0U) yields undefined behavior.
+ * (You may use ilog2(x | 1) to define the (x == 0) behavior.)
+ */
+# define ilog2(x) ((sizeof(x) <= sizeof(unsigned int)) ? \
+     (CHAR_BIT*sizeof(unsigned int)-1-__builtin_clz(x)) : \
+     (sizeof(x) <= sizeof(unsigned long)) ? \
+     (CHAR_BIT*sizeof(unsigned long)-1-__builtin_clzl(x)) : \
+     (CHAR_BIT*sizeof(unsigned long long)-1-__builtin_clzll(x)))
```

Comment:
Use `static inline` and define 3 proper versions.
Also: Does this really have to life in the header or could this be placed into `Meter.c`?

Reply:
@BenBE 
If I remember correctly, htop (at least 2.x) auto-generates header files in build time, so it was not me who can decide which macros reside in the header and which only in .c file.

Also, `ilog2` is an intentionally overloaded function, and many implementations define `ilog2` as a macro. So I followed that way.

Reply:
The automatic generation of header files was removed quite a while back (with 3.0.1 (in early September). And even back when the headers were still auto-generated you had quite a bit of influence as you basically wrote the headers as specially crafted comments in the .c file.

That aside: IMHO `ilog2` should probably go either into a header for utility functions (if other modules can and should make use of it) or reside in the .c file only. As it is right now, it's only used for `Meter.c` and thus is probably suited best there implementation-wise.

On the aspect of `#define` vs `static inline` function: As this implementation has several nested trinary operators, which solely depend on the argument type, it's probably more readable if done explicitly. Also for source included from elsewhere it's nice to have some reference where it came from.

```c
static inline int ilog2u(unsigned int x) {
   return CHAR_BIT*sizeof(x)-1-__builtin_clz(x);
}
static inline int ilog2ul(unsigned long x) {
   return CHAR_BIT*sizeof(x)-1-__builtin_clzl(x);
}
static inline int ilog2ull(unsigned long long x) {
   return CHAR_BIT*sizeof(x)-1-__builtin_clzll(x);
}
```

The only implementation I could find (in a quick search) was the one from the Linux kernel in
[tools/includes/linux/log2.h line 71](https://github.com/torvalds/linux/blob/master/tools/include/linux/log2.h#L71) (which only special-cases numbers up to 4 bytes and those larger than this and apart from this handles the actual implementation as separate functions).